### PR TITLE
Modify `yai` and `newtargets` to accept reduced CCA axes (GNN)

### PR DESCRIPTION
**Pull request by grovduck**
Hi @jeffreyevans @nickcrookston - this PR is meant to accept a user-supplied `nVec` argument to optionally reduce the number of CCA axes used in the "gnn" method.  I'm horrible at R, but I used `yaImpute` for consistency when comparing against [our latest](https://github.com/lemma-osu/scikit-learn-knn-regression) `scikit-learn` based python code.  I found that I had to modify `yai` and `newtargets` to be able to reduce the number of CCA axes (from full rank), but please correct me if I'm wrong.  Please also feel free to ask to modify this code.  There is duplication of code between `yai` and `newtargets`, but that is to avoid modifying the `ccaVegan` object.  I also may not have modified all the locations where this would need to be done - for my comparisons, I only needed to modify in the two locations.

My main reason for adding this is that I feel like "gnn" is being overfit when using all CCA axes.  As far as I can tell, yaImpute uses all CCA axes in prediction, even though later axes typically have very little explanatory power.  We have always truncated GNN output to eight or fewer CCA axes.  This would make "gnn" more similar to "msn" in that "msn" uses `ftest.cor` to set a reasonable number of meaningful axes.  Note that we don't have a similar test for GNN for automatic reduction of axes. 
  
- use `nVec` in `yai` as a user-supplied argument to optionally reduce the number of CCA axes used for imputation
- add `nVec` keyword and apply the same logic in `newtargets`

### CODE REVIEW

**Comment by jeffreyevans** [comment]
Matt, thank you for your time and attention to this. Very good point about gnn over-fitting based on using all of the CCA axis I am wondering about your thoughts on defaulting nVec to 8 rather than NULL? It sounds like it would make things consistent with your scikit-learn code and would be a sensible default, rather that the default behavior allowing over-fit.

**Comment by ghost** [comment]
👍

**Comment by grovduck** [comment]
@jeffreyevans, using `nVec=8` would be just fine.  There is a lame historical precedent for that ... in the early days of when we ran GNN, we used ter Braak's software package `CANOCO` and it would give the user four axes at a time.  So when we initially did a comparison, four axes was too few, twelve axes was too many, and eight axes was the sweet spot.  Not terribly rigorous.

I'm not totally familiar with the f-test that @nickcrookston uses to reduce MSN axes (i.e. `ftest.cor`), but perhaps something similar could be used for GNN?

Also, I'm not sure if `yai` and `newTargets` are the only places that this modification would need to be made.  Those were the functions that I was most familiar with.
